cmd/copyEnumeratorInit.go

```diff
@@ -452,10 +452,11 @@ func (cca *CookedCopyCmdArgs) createDstContainer(containerName string, dstWithSA
 	}
 	existingContainers[containerName] = true
 
-	dstCredInfo := common.CredentialInfo{}
+	var dstCredInfo common.CredentialInfo
 
 	// 3minutes is enough time to list properties of a container, and create new if it does not exist.
-	ctx, _ := context.WithTimeout(parentCtx, time.Minute*3)
+	ctx, cancel := context.WithTimeout(parentCtx, time.Minute*3)
+	defer cancel()
```

Comment:
this is fine right?

Reply:
This means on function return the context will be cancelled. As there is timeout mentioned I feel better to avoid this cancel, as if any go routine is started with the same context, it will get cancelled when these returns.

Reply:
Contexts are an inheritance-type data structure-- They're effectively a linked list @vibhansa-msft. When you say to get a context with a timeout, it does not include the context into the parent, it creates a new context inheriting from the parent, with a timeout. This context is not passed anywhere. It makes sense to cancel it and avoid leaking the ctx for 3 minutes.